File: Problems/No.187_Repeated_DNA_Sequences/repeated_dna_sequences.py
class Solution(object):
    def findRepeatedDnaSequences(self, s):
        """
        :type s: str
        :rtype: List[str]
        """

        # parameter
        DNA_cnt = {}

        # start computing
        if (len(s) <= 10):
            return []
        for i in range(len(s)-10+1):
            sub_DNA = s[i:i+10]
            DNA_cnt[sub_DNA] = DNA_cnt.get(sub_DNA, 0) + 1
        
        return [sDNA for [sDNA, cnt] in DNA_cnt.items() if (cnt > 1)]

File: Problems/No.187_Repeated_DNA_Sequences/repeated_dna_sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        
        // parameters
        std::vector<std::string> RepeatedDNA;
        std::unordered_map<std::string, int> DNA_cnt;

        // start computing
        if (s.length() <= 10) return {};
        for (int i=0; i<=s.length()-10; i++) {

            std::string sub_DNA = s.substr(i, 10);
            DNA_cnt[sub_DNA]++;

        }
        for (auto &[sDNA, cnt]: DNA_cnt)
            if (cnt > 1) RepeatedDNA.push_back(sDNA);
        
        return RepeatedDNA;

    }
};
